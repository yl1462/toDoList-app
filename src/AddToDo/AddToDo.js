import React, { Component } from 'react';
import config from '../config';
import PropTypes from 'prop-types'

class AddToDo extends Component {

  state = {
    title: '',
    description: ''
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  // posting new to do to the page and database
  handleSubmit = (e) => {
    e.preventDefault()
    fetch(`${config.API_ENDPOINT}/api/todo`, {
        method:'POST',
        headers:{
            'Content-Type':'application/json'
        },
        body:JSON.stringify({
            title:this.state.title,
            description: this.state.description,
        })
    })
    .then(res => {
        if (!res.ok) {
            return res.json().then(err => {
                console.log(`Error Message: ${err}`)
                throw err
            })
        }
        return res.json()
    })

    // add to do, then go back to homepage
    .then(todo => {
      this.props.addToDo(todo)
      this.props.history.push('/home')
    })
    .catch(err => {
        this.setState({err})
    })
}

  render() {
    let { title, description } = this.props

    return (
      <>
        <div className="AddToDo">
          <form onSubmit={this.handleSubmit}>

            {/* must give a title to the new to do */}
            <input
              type='text'
              placeholder='title'
              value={title}
              onChange={this.handleChange}
              name='title'
              className="Placeholder"
              required
              pattern="[A-Za-z0-9\~\!\@\#\$\%\^\*\(\)\_]{3}"
              title="at least 3 characters required"
            />
            <br />

            {/* optional description */}
            <textarea
              type='text'
              placeholder='description'
              value={description}
              onChange={this.handleChange}
              name='description'
              className="Placeholder"
            />
            <br />

            {/* submit the new to do, then go back to homepage */}
            <button type='submit' className='AddToDo-button'>Submit</button>
          </form>
        </div>
      </>
    );
  }
}

export default AddToDo;

AddToDo.propTypes = {
  history: PropTypes.object.isRequired
}