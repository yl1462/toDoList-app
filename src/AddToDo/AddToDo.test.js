import React from 'react';
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddToDo from './AddToDo'

describe('renders without crashing', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <AddToDo />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`AddToDo component`, () => {
  const props = {
    title: 'walk the dog',
    description: 'Dogs'
  }

  //test if testing new to do could be added
  it('renders the new todo given props', () => {
    const wrapper = shallow(<AddToDo {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

