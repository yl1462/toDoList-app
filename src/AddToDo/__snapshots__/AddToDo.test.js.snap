// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDo component renders the new todo given props 1`] = `
<Fragment>
  <div
    className="AddToDo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="Placeholder"
        name="title"
        onChange={[Function]}
        placeholder="title"
        required={true}
        type="text"
        value="walk the dog"
      />
      <br />
      <textarea
        className="Placeholder"
        name="description"
        onChange={[Function]}
        placeholder="description"
        type="text"
        value="Dogs"
      />
      <br />
      <button
        className="AddToDo-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</Fragment>
`;
